applications/StructuralMechanicsApplication/tests/cpp_tests/test_truss.cpp

```diff
@@ -0,0 +1,116 @@
+
+namespace Kratos
+{
+namespace Testing
+{
+
+    void AddDisplacementDofsElement(ModelPart& rModelPart){
+        for (auto& r_node : rModelPart.Nodes()){
+            r_node.AddDof(DISPLACEMENT_X);
+            r_node.AddDof(DISPLACEMENT_Y);
+            r_node.AddDof(DISPLACEMENT_Z);
+        }
+    }
+
+    // Tests the mass matrix of the TrussElement3D2N
+    KRATOS_TEST_CASE_IN_SUITE(TrussElement3D2NMassMatrix, KratosStructuralMechanicsFastSuite)
+    {
+        Model current_model;
+        auto &r_model_part = current_model.CreateModelPart("ModelPart",1);
+        r_model_part.GetProcessInfo().SetValue(DOMAIN_SIZE, 3);
+
+        r_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+        r_model_part.AddNodalSolutionStepVariable(VOLUME_ACCELERATION);
+
+        const double density = 7850.0;
+        const double length  = 2.0;
+        const double area = 0.01;
+
+        // Set the element properties
+        auto p_elem_prop = r_model_part.CreateNewProperties(0);
+        p_elem_prop->SetValue(YOUNG_MODULUS, 2.0e+06);
+        p_elem_prop->SetValue(DENSITY, density);
+        p_elem_prop->SetValue(CROSS_AREA, area);
+        const auto &r_clone_cl = KratosComponents<ConstitutiveLaw>::Get("TrussConstitutiveLaw");
+        p_elem_prop->SetValue(CONSTITUTIVE_LAW, r_clone_cl.Clone());
+
+        // Create the test element
+        auto p_node_1 = r_model_part.CreateNewNode(1, 0.0 , 0.0 , 0.0);
+        auto p_node_2 = r_model_part.CreateNewNode(2, length , 0.0 , 0.0);
+
+        AddDisplacementDofsElement(r_model_part);
+
+        std::vector<ModelPart::IndexType> element_nodes {1,2};
+        auto p_element = r_model_part.CreateNewElement("TrussElement3D2N", 1, element_nodes, p_elem_prop);
+
+        const auto& r_process_info = r_model_part.GetProcessInfo();
+
+        p_element->Initialize(r_process_info); // Initialize the element to initialize the constitutive law
+        const auto& r_const_elem_ref = *p_element;
+        r_const_elem_ref.Check(r_process_info);
+
+        const unsigned int number_of_nodes = p_element->GetGeometry().size();
+        const unsigned int dimension = p_element->GetGeometry().WorkingSpaceDimension();
+        const unsigned int number_of_dofs = number_of_nodes * dimension;
+
+
+
+        p_elem_prop->SetValue(COMPUTE_LUMPED_MASS_MATRIX,true);
+        Matrix mm_lumped = ZeroMatrix(number_of_dofs,number_of_dofs);
+        p_element->CalculateMassMatrix(mm_lumped,r_process_info);
+
+        p_elem_prop->SetValue(COMPUTE_LUMPED_MASS_MATRIX,false);
+        Matrix mm_consistent = ZeroMatrix(number_of_dofs,number_of_dofs);
+        p_element->CalculateMassMatrix(mm_consistent,r_process_info);
+
+        for (unsigned int i=0; i<number_of_dofs;++i){
+            double diagonal_entry = 0.0;
+            for (unsigned int j=0; j<number_of_dofs;++j) diagonal_entry += mm_consistent(i,j);
+            KRATOS_CHECK_NEAR(mm_lumped(i,i),diagonal_entry,1.0e-5);
```

Comment:
same

maybe you could also check the function `CalculateLumpedMassVector` since you are already at it, should be very few additional lines